### PR TITLE
Add IoCProvider property to MvxViewModel

### PR DESCRIPTION
**Pull request by borbmizzet**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Adds `IoCProvider` property to `MvxViewModel`, so it can be accessed like `NavigationService` and `Log`.

### :arrow_heading_down: What is the current behavior?
`IMvxIoCProvider` must be passed in view model constructor and stored or use `Mvx.IoCProvider` to reference `IoCProvider` from view models.

### :new: What is the new behavior (if this is a feature change)?
Allows access to `IoCProvider` from view models through base class property.

### :boom: Does this PR introduce a breaking change?
No.

### :bug: Recommendations for testing
Since MvxViewModel.NavigationService is now set by using IoCProvider property, just open a playground project and navigate to another page

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by martijn00** [comment]
I'm thinking we should maybe bring all those extras into 1 interface instead of having a new one for every feature we add.

**Comment by borbmizzet** [comment]
@martijn00 Should that be done in a new PR after merging this one in?

**Comment by borbmizzet** [comment]
seems that second commit broke the tests somehow.  fixing it now

**Comment by borbmizzet** [comment]
The tests still run fine on my machine.  Probably a race condition in the tests.  The changes in my PR should have absolutely no effect on the test that failed.

**Comment by nickrandolph** [comment]
I'm not opposed to having the IMvxIoCViewModel interface but I don't believe the MvxViewModel should expose this as a property - this encourages a very bad coding practice of using the IoCProvider directly to resolve an interface. This is something that should really be avoided.

We currently have code in the properties in MvxViewModel that uses IoCProvider as a fallback, just in case the property isn't set. This code really shouldn't be there - from what I recall, it was added to get the test cases to pass again.

To your point @martijn00, the reason that there are separate interfaces for each of these properties is that they are non-essential and that should someone decide they want to create their own IMvxViewModel, they should be able to do so without have to implement all these additional properties.

In summary:
- I don't believe we should move this forward
- We should work out a better solution, rather than calling Resolve in the properties to enable the test cases to work
- We should remove the fallback calls to Resolve.

**Comment by Cheesebaron** [comment]
I agree with @nickrandolph here. This seems very much like an antipattern. Just like having NavigationService and LogProvider resolved behind your back.

I would much rather prefer to have these as explicit dependencies, rather than implicit and unclear behavior from a public API point of view.

**Comment by nickrandolph** [comment]
@borbmizzet going to close this PR for the moment. We'll revisit how we make it easier for devs to take advantage of core plumbing (eg IoCProvider) going forward
